src/raw_node.rs

```diff
@@ -565,6 +569,11 @@ impl<T: Storage> RawNode<T> {
             }
             let mut record = self.records.pop_front().unwrap();
 
+            if let Some((i, t)) = record.snapshot {
+                index = i;
```

Comment:
Any case to cover?

Reply:
Added by changing `test_async_ready_follower`